## Hypothesis test

I am curious about the length of stay by European people. European countries are near the UK compared to ather region. So, I guess European people spend the UK less than '5' days. Test hypothesis using a 95% confidence interval. 
I will use the data of 2022.

* $H_{0}: \mu \leq 5$
* $H_{1/a}: \mu > 5$

In [2]:
import numpy as np
import pandas as pd

In [3]:
#import the file
data = pd.read_csv('../data/cleaned/data_2022.csv')
data.head()

country  number_of_visit_2022  spending_per_day_2022  length_of_stay_2022  \
0    Canada                   897                  96.77                   11   
1       USA                  4587                 152.05                    9   
2   Austria                   288                 103.82                    6   
3   Belgium                   679                 125.90                    5   
4  Bulgaria                   183                  67.46                    9   

   total_2022_(m) is_EU english_spoken visa_requirement commonwealth  \
0             954    No            Yes               No          Yes   
1            6277    No            Yes               No           No   
2             179   Yes             No               No           No   
3             427   Yes             No               No           No   
4             111   Yes             No               No           No   

   population_(m)                 region         income_group       gdp  
0           38.93          North America          High income  57430.85  
1          333.29          North America          High income  62866.71  
2            9.04  Europe & Central Asia          High income  42400.00  
3           11.67  Europe & Central Asia          High income  39725.42  
4            6.47  Europe & Central Asia  Upper middle income  16750.85

In [7]:
#choose only European people. 
data1 = data[data['region'] == 'Europe & Central Asia']
data1.head()

country  number_of_visit_2022  spending_per_day_2022  \
2         Austria                   288                 103.82   
3         Belgium                   679                 125.90   
4        Bulgaria                   183                  67.46   
5  Czech Republic                   276                 100.49   
6          Cyprus                   123                  80.54   

   length_of_stay_2022  total_2022_(m) is_EU english_spoken visa_requirement  \
2                    6             179   Yes             No               No   
3                    5             427   Yes             No               No   
4                    9             111   Yes             No               No   
5                    5             138   Yes             No               No   
6                   11             108   Yes             No              Yes   

  commonwealth  population_(m)                 region         income_group  \
2           No            9.04  Europe & Central Asia          High income   
3           No           11.67  Europe & Central Asia          High income   
4           No            6.47  Europe & Central Asia  Upper middle income   
5           No           10.53  Europe & Central Asia          High income   
6          Yes            1.25  Europe & Central Asia          High income   

         gdp  
2   42400.00  
3   39725.42  
4   16750.85  
5  505259.65  
6   26653.36

In [10]:
import scipy.stats as st
from scipy.stats import ttest_1samp

In [11]:

length = data1['length_of_stay_2022']
sample_mean = length.mean()
sample_std = length.std()
sample_size = len(length)
Population_mean = 5


In [15]:
# One-sided test:
# Null hypothesis or H0: mean cholesterol value <= 5
# Alternative hyp or H1: mean cholesterol value > 5

stat, pval = ttest_1samp(length, popmean = 5, alternative = "greater")
print('Ttest_1sampResult(statistic={}, pvalue={})'.format(stat, pval))


Ttest_1sampResult(statistic=4.241987828412078, pvalue=0.0001327917536294034)


In this case the p_value is bigger than $\alpha=0.05$. Therefore we **reject the null hypothesis**

Manual Calculation

In [17]:
critical_value = st.t.ppf(1-0.05, df=len(length)-1)
print("The critical value is {:.3f}".format(critical_value))

The critical value is 1.708


In our case, we have:

$$1.708 < 4.241$$

Therefore, we **reject the null hypothesis** 

### Chi2 test

#### Question: Are Visa requirement and income group releted or not?

In [21]:
data_crosstab = pd.crosstab(data['visa_requirement'], data['income_group'], margins = False)
data_crosstab

income_group      High income  Lower middle income  Upper middle income
visa_requirement                                                       
No                         31                    2                    5
Yes                         1                    5                    4

* $H_{0}:  independent$
* $H_{1/a}: not \_independent$

In [23]:
# (chi2_statistic, p_value, degrees_of_freedom, expected_frequencies_array) = st.chi2_contingency(contingency table)
st.chi2_contingency(np.array(data_crosstab))

(19.990977443609026,
 4.560520415249994e-05,
 2,
 array([[25.33333333,  5.54166667,  7.125     ],
        [ 6.66666667,  1.45833333,  1.875     ]]))

In [24]:
data_crosstab.shape

(2, 3)

In [26]:
critical_value2= st.chi2.ppf(1-0.05,df=2)
print("The critical value is {:.3f}".format(critical_value2))

The critical value is 5.991


In our case, we have:

$$19.990 > 5.991$$ 

P-value: 4.560520415249994e-05.

$$4.560520415249994e-05 < 0.05$$ 

Therefore, we **reject the null hypothesis** 